### PR TITLE
core/corehttp!: remove /api/v0 from gateway port

### LINKED ISSUE #10312: Deprecate and remove `/api/v0` under the gateway port
We've been wanting to deprecate the access to `/api/v0` under the gateway path for [at least two years](https://github.com/ipfs/kubo/commit/84ed0ec59f2472e8ca4ef7f5f3f4027fe58b10ac#diff-28221ed38896d1567acc67ba8c040755be2931ff71190fc4efd2f3e8c7b94d91R116), but I have not found any issue. This is the tracking issue.

- [x] **Kubo 0.27:** deprecate `/api/v0` from gateway port: https://github.com/ipfs/kubo/pull/10311.
- [ ] **Kubo 0.28**: completely remove `/api/v0` from gateway port: https://github.com/ipfs/kubo/pull/10352

### PR DESCRIPTION
**Pull request by hacdias**
Removes `/api/v0` from the gateway port. Closes #10312.

We can merge this before the final 0.27 release. I will have to cherry-pick anyways.

### CODE REVIEW

File: docs/changelogs/v0.28.md [hunk 1]
```diff
@@ -6,12 +6,17 @@
...
 - [👨‍👩‍👧‍👦 Contributors](#-contributors)
 
 ### Overview
 
-### 🔦 Highlights
+#### Gateway: `/api/v0` is removed
+
+The legacy subset of the Kubo RPC that was available via the Gateway port and was deprecated is now completely removed. You can read more in <https://github.com/ipfs/kubo/issues/10312>.
+
+If you have a legacy software that relies on this behavior, and want to expose parts of `/api/v0` next to `/ipfs`, use reverse-proxy in front of Kubo to mount both Gateway and RPC on the same port. NOTE: exposing RPC to the internet comes with security risk: make sure to specify access control via [API.Authorizations](https://github.com/ipfs/kubo/blob/master/docs/config.md#apiauthorizations).
```
Comment (lidel) [inline docs/changelogs/v0.28.md:19]: :ok_hand:

File: test/sharness/t0002-docker-image.sh [hunk 2]
```diff
@@ -50,7 +50,7 @@ test_expect_success "docker image runs" '
 '
 
 test_expect_success "docker container gateway is up" '
-  pollEndpoint -host=/ip4/127.0.0.1/tcp/8080 -http-url http://localhost:8080/api/v0/version -v -tries 30 -tout 1s
```
Comment (lidel) [inline test/sharness/t0002-docker-image.sh:53]: :thought_balloon: I suspect this is used by various monitoring tools.
Mind updating waterworks-infra in preparation for this removal?

`/api/v0/version` seems to be used in a few places, including pingdom:

https://github.com/search?q=repo%3Aipshipyard%2Fwaterworks-infra+api%2Fv0%2Fversion&type=code

Maybe @ns4plabs knows other places.

File: core/commands/commands_test.go [hunk 3]
```diff
@@ -15,63 +15,6 @@ func collectPaths(prefix string, cmd *cmds.Command, out map[string]struct{}) {
...
-		"/ls",
-		"/name",
-		"/name/resolve",
-		"/object",
-		"/object/data",
-		"/object/get",
-		"/object/links",
-		"/object/stat",
-		"/refs",
-		"/resolve",
```
Comment (lidel) [inline core/commands/commands_test.go:44]: :thought_balloon: if someone needs replacement for `/api/v0/resolve` we could tell them to use `x-ipfs-roots`:

```console
$ curl https://ipfs.io/ipns/en.wikipedia-on-ipfs.org/favicon.ico -X HEAD -is | grep -i ipfs-roots
x-ipfs-roots: bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze,bafkreicl5473u3g5c2l34n5oh5trtpft56ep6mbqv57affzgkfppcitfl4
```

**Review by lidel — APPROVED** [review]
Thank you for cleaning this up. Long time due.
Removing Kubo RPC from Gateway port should reduce confusion in the long run.

File: test/sharness/t0002-docker-image.sh (hunk 2 above)
Comment (hacdias) [reply test/sharness/t0002-docker-image.sh:53]: https://github.com/ipshipyard/waterworks-infra/pull/41

**Pull request merged**